tests/integration-tests/src/commonTest/kotlin/test/WatcherTest.kt

```diff
@@ -74,19 +81,26 @@ class WatcherTest {
   fun sameQueryTriggersWatcher() = runTest(before = { setUp() }) {
     val query = EpisodeHeroNameQuery(Episode.EMPIRE)
     val channel = Channel<EpisodeHeroNameQuery.Data?>()
+    val channel2 = Channel<EpisodeHeroNameQuery.Data?>()
 
     // The first query should get a "R2-D2" name
-    apolloClient.enqueueTestResponse(query, episodeHeroNameData)
+    mapResponses.register(query, ApolloResponse.Builder(query, uuid4(), episodeHeroNameData).build())
     val job = launch {
       apolloClient.query(query).watch().collect {
         channel.send(it.data)
       }
     }
+    val job2 = launch {
+      apolloClient.query(query).watch().collect {
+        channel2.send(it.data)
+      }
+    }
 
     assertEquals(channel.receiveOrTimeout()?.hero?.name, "R2-D2")
+    assertEquals(channel2.receiveOrTimeout()?.hero?.name, "R2-D2")
 
     // Another newer call gets updated information with "Artoo"
-    apolloClient.enqueueTestResponse(query, episodeHeroNameChangedData)
+    mapResponses.register(query, ApolloResponse.Builder(query, uuid4(), episodeHeroNameChangedData).build())
```

Comment:
In a similar vein as what @martinbonnin tried yesterday, if I add a single `yield()` right before this line, the test passes for me. Of course this doesn't really help explaining what's wrong... 😅

Reply:
haha yeah I still don't know if it's a unit test environment issue or a real issue that could happen on a real device 🤔